internal/static/cartogram2.js

```diff
@@ -3326,190 +3401,226 @@ class Cartogram {
         {
             if(document.getElementById('csv').files.length < 1)
             {
-                this.doNonFatalError(Error('You must upload CSV data.'));
+                this.doNonFatalError(Error('You must upload CSV/Excel data.'));
                 return false;
             }
+            
+            // convert given xlsx file to csv
+            var input_data_file = document.getElementById('csv').files[0];
+            if(input_data_file.name.split('.').pop() === 'xlsx') {
```

Comment:
Does your `XLSX` library support reading old-style `.xls` files too? If so, you may want to add support for them here.